docs/developer/tutorials/dev-install.rst

```diff
@@ -0,0 +1,45 @@
+Developer install
+=================
+
+These instructions will take you through the minimal steps required to get a dev
+environment setup, so you can run the tests locally.
+
+Clone the repository
+--------------------
+
+First clone the repository locally using `Git
+<https://git-scm.com/downloads>`_::
+
+    $ git clone git://github.com/bluesky/event-model.git
+
+Install dependencies
+--------------------
+
+You should install into a `venv` (which requires python 3.8 or later):
+.. code::
+
+    $ cd event-model
+    $ python3 -m venv venv
+    $ source venv/bin/activate
+    $ pip install -e '.[dev]'
+
+See what was installed
+----------------------
+
+To see a graph of the python package dependency tree type::
+
+    $ pipdeptree
+
+Build and test
+--------------
+
+Now you have a development environment you can run the tests in a terminal::
+
+    $ tox -p
```

Comment:
I ran it locally, and there was an issue with the `pre-commit` run. There were errors complaining about Python 3.1 (looks like the `3.10` version was used as a float, not string somewhere):
```
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/Users/mrakitin/anaconda3/envs/bluesky/bin/python3.1', '-mvirtualenv', '/Users/mrakitin/.cache/pre-commit/repob5ocz7s_/py_env-python3.1', '-p', 'python3.1')
return code: 1
expected return code: 0
stdout:
    RuntimeError: failed to find interpreter for Builtin discover of python_spec='python3.1'
```
Otherwise, the tests ran fine.

Reply:
That's odd, do you have a set of steps to reproduce from a fresh clone? I can't see any obvious places it could have used python version as a float...

Reply:
I am using conda, and I guess it fails to create a venv properly within a conda env. I had to manually fix the shebang line in the `pre-commit` script to point to the right version of Python.

I've rerun the installation on another machine in a clean conda env (created with `conda create -n event-model python=3.10 -c conda-forge -y`). The problem does not appear there. I guess it was a local cache issue or something similar.